File: Flutter_Client/bicho_pedia/lib/auth/models/register_dto.dart
class RegisterDto {
  String? username;
  String? email;
  String? password;

  RegisterDto({this.username, this.email, this.password});

  RegisterDto.fromJson(Map<String, dynamic> json) {
    username = json['username'];
    email = json['email'];
    password = json['password'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['username'] = this.username;
    data['email'] = this.email;
    data['password'] = this.password;
    return data;
  }
}

File: API/bichoapi/src/main/java/com/robertorebolledonaharro/bichoapi/user/dto/RegisterDTO.java
package com.robertorebolledonaharro.bichoapi.user.dto;

import com.robertorebolledonaharro.bichoapi.user.validator.ValidPassword;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

@ValidPassword
public record RegisterDTO(

        @NotNull(message = "username can`t be null")
        @NotBlank(message = "username can`t be black")
        String username,

        @NotNull(message = "email can`t be null")
        String email,

        String password



) {



}
